tools/org.eclipse.n4js.hlc.base/src/org/eclipse/n4js/hlc/base/N4jscBase.java

```diff
@@ -381,6 +381,15 @@ public SuccessExitStatus doMain(String... args) throws ExitCodeException {
 				return SuccessExitStatus.INSTANCE;
 			}
 
+			// Make sure the srcFiles are valid
+			for (File srcFile : srcFiles) {
+				if (!srcFile.isFile() && !srcFile.isDirectory()) {
+					System.out.println(srcFile.toString()
+							+ " of the list of source files or projects is neither valid file nor valid directory.");
+					throw new ExitCodeException(ErrorExitCode.EXITCODE_SRCFILES_INVALID);
```

Comment:
consider checking all files before throwing the exception and then including all invalid files in the message

Reply:
Done.